privilege/privileges/privileges_test.go

```diff
@@ -843,330 +820,387 @@ func connectionState(issuer, subject pkix.Name, cipher uint16, opt ...func(c *x5
 	_, err = se.ExecuteInternal(context.Background(), "analyze table t1")
-	c.Assert(terror.ErrorEqual(err, core.ErrTableaccessDenied), IsTrue)
-	c.Assert(err.Error(), Equals, "[planner:1142]INSERT command denied to user 'anobody'@'%' for table 't1'")
+	require.True(t, terror.ErrorEqual(err, core.ErrTableaccessDenied))
+	require.EqualError(t, err, "[planner:1142]INSERT command denied to user 'anobody'@'%' for table 't1'")
 	// Add INSERT privilege and it should work.
-	c.Assert(se.Auth(&auth.UserIdentity{Username: "asuper", Hostname: "localhost", AuthUsername: "asuper", AuthHostname: "%"}, nil, nil), IsTrue)
-	mustExec(c, se, "GRANT INSERT ON atest.* TO 'anobody'")
-	c.Assert(se.Auth(&auth.UserIdentity{Username: "anobody", Hostname: "localhost", AuthUsername: "anobody", AuthHostname: "%"}, nil, nil), IsTrue)
+	require.True(t, se.Auth(&auth.UserIdentity{Username: "asuper", Hostname: "localhost", AuthUsername: "asuper", AuthHostname: "%"}, nil, nil))
+	mustExec(t, se, "GRANT INSERT ON atest.* TO 'anobody'")
+	require.True(t, se.Auth(&auth.UserIdentity{Username: "anobody", Hostname: "localhost", AuthUsername: "anobody", AuthHostname: "%"}, nil, nil))
 	_, err = se.ExecuteInternal(context.Background(), "analyze table t1")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 }
 
-func (s *testPrivilegeSuite) TestSystemSchema(c *C) {
+func TestSystemSchema(t *testing.T) {
+	t.Parallel()
+	store, clean := newStore(t)
+	defer clean()
+
 	// This test tests no privilege check for INFORMATION_SCHEMA database.
-	se := newSession(c, s.store, s.dbName)
-	mustExec(c, se, `CREATE USER 'u1'@'localhost';`)
-	c.Assert(se.Auth(&auth.UserIdentity{Username: "u1", Hostname: "localhost"}, nil, nil), IsTrue)
-	mustExec(c, se, `select * from information_schema.tables`)
-	mustExec(c, se, `select * from information_schema.key_column_usage`)
+	se := newSession(t, store, dbName)
+	mustExec(t, se, `CREATE USER 'u1'@'localhost';`)
+	require.True(t, se.Auth(&auth.UserIdentity{Username: "u1", Hostname: "localhost"}, nil, nil))
+	mustExec(t, se, `select * from information_schema.tables`)
+	mustExec(t, se, `select * from information_schema.key_column_usage`)
 	_, err := se.ExecuteInternal(context.Background(), "create table information_schema.t(a int)")
-	c.Assert(strings.Contains(err.Error(), "denied to user"), IsTrue)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "denied to user"))
 	_, err = se.ExecuteInternal(context.Background(), "drop table information_schema.tables")
-	c.Assert(strings.Contains(err.Error(), "denied to user"), IsTrue)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "denied to user"))
 	_, err = se.ExecuteInternal(context.Background(), "update information_schema.tables set table_name = 'tst' where table_name = 'mysql'")
-	c.Assert(strings.Contains(err.Error(), "privilege check"), IsTrue)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "privilege check"))
```

Comment:
`require.Contains`?

Reply:
Maybe use `require.True(t, terror.ErrorEqual(err, core.ErrPrivilegeCheckFail))` instead?

Reply:
It is even better!